ctapipe/io/magiceventsource.py

```diff
@@ -0,0 +1,1352 @@
+
+
+import glob
+import re
+
+import h5py
+import numpy as np
+import scipy.interpolate
+
+from astropy import units as u
+from astropy.coordinates import Angle
+from astropy.time import Time
+from ctapipe.io.eventsource import EventSource
+from ctapipe.io.containers import DataContainer, TelescopePointingContainer, WeatherContainer
+from ctapipe.instrument import TelescopeDescription, SubarrayDescription, OpticsDescription, CameraGeometry
+import gzip
+import struct
+
+
+__all__ = ['MAGICEventSourceHDF5', 'MAGICEventSourceROOT']
+
+
+class MAGICEventSourceHDF5(EventSource):
+    """
+    EventSource for MAGIC raw data converted to hdf5.
+
+    This class utilises `h5py` to read the hdf5 file, and stores the
+    information into the event containers.
+    """
+    _count = 0
+
+    def __init__(self, config=None, tool=None, **kwargs):
+        super().__init__(config=config, tool=tool, **kwargs)
+
+        self.h5py = h5py
+        self.file = h5py.File(self.input_url)
+
+    @staticmethod
+    def is_compatible(file_path):
+        import h5py
+        # check general format:
+        if not h5py.is_hdf5(file_path):
+            return False
+        # crude check if hdf5 file contains MAGIC raw data:
+        with h5py.File(file_path, "r") as file:
+            intrument_attr = file.attrs['instrument']
+            if intrument_attr.tostring() == b"MAGIC":
+                return True
+            else:
+                return False
+            
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.file.close()
+
+    def _generator(self):
```

Comment:
I just learned, that this should be used by users as public API, so it should not start with an `_`.

Maybe a fitting name would be `iter_stereo_events`